### PR TITLE
CLI: Allow spaces in -p arguments

### PR DESCRIPTION
**Pull request by k4nar**
Most pytest CLI arguments expecting a value accept the form `--arg=value`, which
makes it easy to have a single line per argument in the `addopts = [...]` section of
the configuration.

This is not the case for `-p` (to enable or disable pluggins), because it is
parsed in a different step. This argument must either consist of two strings
(`"-p", "<value>"`), or a single string without space (`"-p<value>"`). This is
perfectly fine for the CLI, but less intuitive for the config file. This PR
allows `"-p <value>"` to work as well.

Basically, this PR makes the following config work:
```toml
[tool.pytest.ini_options]
addopts = [
    # ...
    "-p no:logging",
    # ...
]
```

Note that this is mostly an esthetic change, the behavior remains the same.

### CODE REVIEW

**Review by nicoddemus — APPROVED** [review]
Awesome, thanks!
